src/manifest.c

```diff
@@ -713,9 +786,11 @@ struct manifest *load_manifest(int current, int version, struct file *file, stru
 		}
 		return NULL;
 	}
-	retried = false;
+	if (!retried) {
+		retried = false;
```

Comment:
+1 to the query. I also think the variable needs to be renamed, perhaps to "partial_cleanup_needed"?
The whole function could do with being rewritten to remove all the retry logic and putting the retry logic in a new function.